app/routes.py

```diff
@@ -1,2 +1,239 @@
+            for task_id in request_body["task_ids"]:
+                tasks_instances.append(Task.query.get(task_id))
+            
+            goal.tasks = tasks_instances
+            
+            db.session.commit()
+
+            task_ids = []
+            for task in goal.tasks:
+                task_ids.append(task.task_id)
+             
+            response_body = {
+                        "id": goal.goal_id,
+                        "task_ids": task_ids
+                    }
+                    
+            return jsonify(response_body), 200
+
+        if request.method == "GET":
+            tasks_response =[]
+            for task in goal.tasks:
+                tasks_response.append({
+                    "id": task.task_id,
+                    "goal_id": task.goal_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": bool(task.completed_at)
+                })
+            response_body = {
+                "id": goal.goal_id,
+                "title": goal.title,
+                "tasks" : tasks_response
+            }
+            return jsonify(response_body), 200
+
+
+@tasks_bp.route("",methods=["GET","POST"])
+def handle_task():
+    if request.method == "GET": 
+
+        sorting_task = request.args.get('sort') 
+        list = None
+        if sorting_task == "desc":
+            list = Task.query.order_by(Task.title.desc()) # descending method
+        elif sorting_task == "asc":
+            list = Task.query.order_by(Task.title.asc()) # ascending method
+        else: 
+            list = Task.query.all()
+        tasks_response = []
+        for task in list:
+            tasks_response.append({
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete" : False
+            if task.completed_at == None else True
+            })
+
+        return jsonify(tasks_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+            return{
+                "details": "Invalid data"
+            },400
+
+        new_task = Task (
+            title=request_body["title"],
+            description=request_body["description"],
+            completed_at=request_body["completed_at"]
+        )
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        return {
+            "task":new_task.task_dict()
+        }, 201
+
+def update_completion(task_id, value):
+    task = Task.query.get(task_id)
+    
+    if not task:
+        return("Task not found",404)
+    
+    task.completed_at = value 
+
+    return {
+        "task":task.task_dict()
+    }, 200 
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    return update_completion(task_id, datetime.now())
```

Comment:
Consider using `utcnow()` rather than `now()`, especially if there's no timezone information on the datetime in the database. When working with dates and time, it's usually a good idea to pass them around and store them in UTC, and only convert them to local time when needed for display (like in a UI). Working with dates and times is tricky!